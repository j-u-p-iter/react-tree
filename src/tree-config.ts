import { IConfigItem } from "./Tree";

export const treeConfig: IConfigItem[] = [
  {
    id: "1",
    title: "Title 1",
    children: [
      {
        id: "2",
        title: "Title 2",
        children: [
          {
            id: "3",
            title: "Title 3",
            children: [
              {
                id: "4",
                title: "Title 4",
                children: [
                  {
                    id: "5",
                    title: "Title 5"
                  },
                  {
                    id: "6",
                    title: "Title 6"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  },
  {
    id: "7",
    title: "Title 7",
    children: [
      {
        id: "8",
        title: "Title 8",
        children: [
          {
            id: "9",
            title: "Title 9"
          }
        ]
      }
    ]
  }
];
