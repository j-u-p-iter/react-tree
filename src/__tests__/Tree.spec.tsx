import * as React from "react";
import { cleanup, fireEvent, render } from "react-testing-library";

import { Tree } from "../.";
import { treeConfig } from "../tree-config";

describe("Tree", () => {
  let getTreeEl: any;

  beforeAll(() => {
    getTreeEl = (props: any = {}) => (
      <Tree config={treeConfig} {...props}>
        {({ api: { toggleNode, getTree } }) => {
          const renderTree = (tree: any) =>
            tree.map((node: any) => {
              return (
                <li
                  key={node.id}
                  id={node.id}
                  data-active={node.active}
                  onClick={toggleNode}
                >
                  {node.title}
                  {node.children ? <ul>{renderTree(node.children)}</ul> : null}
                </li>
              );
            });

          return <ul>{renderTree(getTree())}</ul>;
        }}
      </Tree>
    );
  });

  afterEach(cleanup);

  it("renders correctly by default", () => {
    const { container } = render(getTreeEl());

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with activeId", () => {
    const { container } = render(getTreeEl({ activeId: "2" }));

    expect(container.firstChild).toMatchSnapshot();
  });

  it("toggles nodes correctly", () => {
    const { container, getByText } = render(getTreeEl({ activeId: "2" }));

    expect(container.firstChild).toMatchSnapshot();

    // opens node with "Title 3"
    fireEvent.click(getByText("Title 3"));
    expect(container.firstChild).toMatchSnapshot();

    // closes node with "Title 2"
    fireEvent.click(getByText("Title 2"));
    expect(container.firstChild).toMatchSnapshot();
  });
});
