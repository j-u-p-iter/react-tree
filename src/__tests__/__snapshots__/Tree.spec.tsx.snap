// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree renders correctly by default 1`] = `
<ul>
  <li
    data-active="false"
    id="1"
  >
    Title 1
    <ul>
      <li
        data-active="false"
        id="2"
      >
        Title 2
        <ul>
          <li
            data-active="false"
            id="3"
          >
            Title 3
            <ul>
              <li
                data-active="false"
                id="4"
              >
                Title 4
                <ul>
                  <li
                    data-active="false"
                    id="5"
                  >
                    Title 5
                  </li>
                  <li
                    data-active="false"
                    id="6"
                  >
                    Title 6
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    data-active="false"
    id="7"
  >
    Title 7
    <ul>
      <li
        data-active="false"
        id="8"
      >
        Title 8
        <ul>
          <li
            data-active="false"
            id="9"
          >
            Title 9
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`Tree renders correctly with activeId 1`] = `
<ul>
  <li
    data-active="true"
    id="1"
  >
    Title 1
    <ul>
      <li
        data-active="true"
        id="2"
      >
        Title 2
        <ul>
          <li
            data-active="false"
            id="3"
          >
            Title 3
            <ul>
              <li
                data-active="false"
                id="4"
              >
                Title 4
                <ul>
                  <li
                    data-active="false"
                    id="5"
                  >
                    Title 5
                  </li>
                  <li
                    data-active="false"
                    id="6"
                  >
                    Title 6
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    data-active="false"
    id="7"
  >
    Title 7
    <ul>
      <li
        data-active="false"
        id="8"
      >
        Title 8
        <ul>
          <li
            data-active="false"
            id="9"
          >
            Title 9
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`Tree toggles nodes correctly 1`] = `
<ul>
  <li
    data-active="true"
    id="1"
  >
    Title 1
    <ul>
      <li
        data-active="true"
        id="2"
      >
        Title 2
        <ul>
          <li
            data-active="false"
            id="3"
          >
            Title 3
            <ul>
              <li
                data-active="false"
                id="4"
              >
                Title 4
                <ul>
                  <li
                    data-active="false"
                    id="5"
                  >
                    Title 5
                  </li>
                  <li
                    data-active="false"
                    id="6"
                  >
                    Title 6
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    data-active="false"
    id="7"
  >
    Title 7
    <ul>
      <li
        data-active="false"
        id="8"
      >
        Title 8
        <ul>
          <li
            data-active="false"
            id="9"
          >
            Title 9
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`Tree toggles nodes correctly 2`] = `
<ul>
  <li
    data-active="true"
    id="1"
  >
    Title 1
    <ul>
      <li
        data-active="true"
        id="2"
      >
        Title 2
        <ul>
          <li
            data-active="true"
            id="3"
          >
            Title 3
            <ul>
              <li
                data-active="false"
                id="4"
              >
                Title 4
                <ul>
                  <li
                    data-active="false"
                    id="5"
                  >
                    Title 5
                  </li>
                  <li
                    data-active="false"
                    id="6"
                  >
                    Title 6
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    data-active="false"
    id="7"
  >
    Title 7
    <ul>
      <li
        data-active="false"
        id="8"
      >
        Title 8
        <ul>
          <li
            data-active="false"
            id="9"
          >
            Title 9
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;

exports[`Tree toggles nodes correctly 3`] = `
<ul>
  <li
    data-active="true"
    id="1"
  >
    Title 1
    <ul>
      <li
        data-active="false"
        id="2"
      >
        Title 2
        <ul>
          <li
            data-active="true"
            id="3"
          >
            Title 3
            <ul>
              <li
                data-active="false"
                id="4"
              >
                Title 4
                <ul>
                  <li
                    data-active="false"
                    id="5"
                  >
                    Title 5
                  </li>
                  <li
                    data-active="false"
                    id="6"
                  >
                    Title 6
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    data-active="false"
    id="7"
  >
    Title 7
    <ul>
      <li
        data-active="false"
        id="8"
      >
        Title 8
        <ul>
          <li
            data-active="false"
            id="9"
          >
            Title 9
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>
`;
